React dev doc dark theme

huayunh opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->

There is no JIRA issue associated with this.

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->

## Changes proposed in this Pull Request:

- Added a theme toggler
- Ensured that most of the UI work in the dark theme

<!-- Include screenshots if they will help illustrate the changes in this PR -->

## Screenshots / Screen Recording (if applicable)

![image](https://user-images.githubusercontent.com/8997218/229940739-1faa73ab-3215-4242-b305-a2ddefa2d480.png)

## To Test

- [Toggle light/dark theme in Chrome's developer tool](https://stackoverflow.com/questions/57606960/how-can-i-emulate-prefers-color-scheme-media-query-in-chrome)
- Known issue: when switching between light and dark theme, the underline in a tab always gets misaligned, like this: 
![image](https://user-images.githubusercontent.com/8997218/229941692-fc09e6a5-3043-4bca-8fee-c9f878646985.png)
- Known issue: https://github.com/etn-ccis/blui-react-themes/issues/66

surajeaton commented on docs/src/shared/components/InfoCard/InfoCard.tsx:
```diff
@@ -97,7 +97,9 @@ export const InfoCard: React.FC<InfoCardProps> = (props): JSX.Element => {
+            <Typography variant={'h6'} color={'text.primary'}>
+                {props.title}
+            </Typography>
             <Typography variant={'body2'} style={{ color: theme.palette.text.secondary, marginTop: theme.spacing(1) }}>
```
I missed this while migrating from doc-it. Can you please update this to have sx prop:
`sx={{color: 'text.secondary', mt: 1}}`

surajeaton reviewed (commented):
I see the different styles in the react icon at left top drawer header while in the light and dark themes:

https://user-images.githubusercontent.com/120575281/230007104-d6793746-ece7-461c-9120-04f2cd0713c4.mov

huayunh commented:
Thank you, @surajeaton . I have addressed all the concerns you've raised. The chips shifting around is a react theme thing — I have logged an issue [here](https://github.com/etn-ccis/blui-react-themes/issues/66). Feel free to take another pass at this.

JeffGreiner-eaton commented:
in dark theme the mui-backdrop is being applied when it is set to invisible. Might be a zIndex thing

![image](https://user-images.githubusercontent.com/119693939/230179034-37a00182-3493-48eb-a8bd-2d86eed3b164.png)

JeffGreiner-eaton commented:
are there plans for system default or can it br removed?
![image](https://user-images.githubusercontent.com/119693939/230179610-a83c1eda-5ce7-494a-9742-9237df7eb611.png)

JeffGreiner-eaton commented:
playground scoreCard looks odd.
![image](https://user-images.githubusercontent.com/119693939/230179986-407eac99-abaf-4aa6-bb7e-5fa397d5c787.png)

huayunh commented:
> are there plans for system default or can it be removed?

"system default" is whatever your browser theme is set to be. And your browser is most likely following the preference setting of your operating system. So if your macOS is set to be the dark theme, with "system default" selected, you should see the dark theme used in the application.

huayunh commented:
> playground scoreCard looks odd.

There are quite a few components with strange playground dark theme colors. These changes are better addressed at the component level — for example, by allowing users to pass `"background.paper"` to a `backgroundColor` prop. Therefore I left them the way they are.

huayunh commented:
I am totally open to new naming suggestions on that.

surajeaton reviewed (approved):
Good job @huayunh.
